src/loaders/artifacts.js

```diff
@@ -1,32 +1,75 @@
 import DataLoader from 'dataloader';
 import sift from 'sift';
+import { isNil } from 'ramda';
+import { withRootUrl } from 'taskcluster-lib-urls';
 import ConnectionLoader from '../ConnectionLoader';
 import Artifact from '../entities/Artifact';
 import Artifacts from '../entities/Artifacts';
 
-export default ({ queue }, isAuthed) => {
-  const withUrl = (taskId, runId, artifact) => {
-    const url = isAuthed
-      ? queue.buildSignedUrl(queue.getArtifact, taskId, runId, artifact.name)
-      : queue.buildUrl(queue.getArtifact, taskId, runId, artifact.name);
+export default ({ queue }, isAuthed, rootUrl) => {
+  const urls = withRootUrl(rootUrl);
+  const withUrl = ({ method, taskId, artifact, runId }) => {
+    const hasRunId = !isNil(runId);
+    const isPublicLog = /^public\/logs\//.test(artifact.name);
 
-    return { ...artifact, url };
+    // We don't want to build signed URLs for public artifacts,
+    // even when credentials are present -- users often
+    // copy/paste artifact URLs, and this would likely lead to
+    // people copy/pasting time-limited, signed URLs which would
+    // (a) have a long ?bewit=.. argument and
+    // (b) probably not work after that bewit expires.
+    // We could use queue.buildUrl, but this creates URLs where the artifact
+    // name has slashes encoded. For artifacts we specifically allow slashes
+    // in the name unencoded, as this make things like `wget ${URL}` create
+    // files with nice names.
```

Comment:
I had forgotten about that -- good catch!  Fixing that in the client might also make a good "good-first-bug" bug..